### DISCUSSION TITLE
Spawned task panics instead of returning error in JoinHandle

### DISCUSSION THREAD

**Discussion by kevin-june** [tokio-rs/tokio#5624] [Q&A] [answered]
Hi all,

One of the [doc examples](https://docs.rs/tokio/latest/tokio/task/struct.JoinHandle.html#examples) isn't behaving how I expect: it panics when I expect it to return a `JoinError`.

Here is the example code:

```rs
use tokio::task;
use std::io;
use std::panic;

#[tokio::main]
async fn main() -> io::Result<()> {
    let join_handle: task::JoinHandle<Result<i32, io::Error>> = tokio::spawn(async {
        panic!("boom");
    });

    let err = join_handle.await.unwrap_err();
    assert!(err.is_panic());
    Ok(())
}
```

The docs state that:

> If the task panics, the error is a [JoinError](https://docs.rs/tokio/latest/tokio/task/struct.JoinError.html) that contains the panic

However, when I run the example code locally or in the Rust playground, I see the panic crash the entire program.

Am I misunderstanding the docs? Perhaps there is some configuration that I am missing? Here is a link to a [playground](https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=daa7cec899b41942c8a21a9702180176) that illustrates this behavior.

Thanks!

**Comment by Darksonn** [comment, marked as answer]
The panic message is printed even if you catch it. Try this:
```Rust
use tokio::task;
use std::io;
use std::panic;

#[tokio::main]
async fn main() -> io::Result<()> {
    let join_handle: task::JoinHandle<Result<i32, io::Error>> = tokio::spawn(async {
        panic!("boom");
    });

    let err = join_handle.await.unwrap_err();
    assert!(err.is_panic());

    println!("at the end");

    Ok(())
}
```

**Reply by kevin-june** [reply]
Oh! Thank you. That is true.

I didn't notice that the program ran to completion, but now that I look more closely I see that the exit code is `0` (success).

Thank you for your time!
